rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -50,91 +53,95 @@ class CountVectorsFeaturizer(SparseFeaturizer):
+            "max_features": None,  # int or None
+            # if convert all characters to lowercase
+            "lowercase": True,  # bool
+            # handling Out-Of-Vocabulary (OOV) words
+            # will be converted to lowercase if lowercase is True
+            "OOV_token": None,  # string or None
+            "OOV_words": [],  # string or list of strings
+            # indicates whether the featurizer should use the lemma of a word for
+            # counting (if available) or not
+            "use_lemma": True,
+            # Additional vocabulary size to be kept reserved for finetuning
+            "additional_vocabulary_size": {
+                TEXT: None,
+                RESPONSE: None,
+                ACTION_TEXT: None,
+            },
+        }
 
-    @classmethod
-    def required_packages(cls) -> List[Text]:
+    @staticmethod
+    def required_packages() -> List[Text]:
         return ["sklearn"]
 
     def _load_count_vect_params(self) -> None:
 
         # Use shared vocabulary between text and all other attributes of Message
-        self.use_shared_vocab = self.component_config["use_shared_vocab"]
+        self.use_shared_vocab = self._config["use_shared_vocab"]
 
         # set analyzer
-        self.analyzer = self.component_config["analyzer"]
+        self.analyzer = self._config["analyzer"]
 
         # remove accents during the preprocessing step
-        self.strip_accents = self.component_config["strip_accents"]
+        self.strip_accents = self._config["strip_accents"]
 
         # list of stop words
-        self.stop_words = self.component_config["stop_words"]
+        self.stop_words = self._config["stop_words"]
 
         # min number of word occurancies in the document to add to vocabulary
-        self.min_df = self.component_config["min_df"]
+        self.min_df = self._config["min_df"]
 
         # max number (fraction if float) of word occurancies
         # in the document to add to vocabulary
-        self.max_df = self.component_config["max_df"]
+        self.max_df = self._config["max_df"]
 
         # set ngram range
-        self.min_ngram = self.component_config["min_ngram"]
-        self.max_ngram = self.component_config["max_ngram"]
+        self.min_ngram = self._config["min_ngram"]
+        self.max_ngram = self._config["max_ngram"]
 
         # limit vocabulary size
-        self.max_features = self.component_config["max_features"]
+        self.max_features = self._config["max_features"]
 
         # if convert all characters to lowercase
-        self.lowercase = self.component_config["lowercase"]
+        self.lowercase = self._config["lowercase"]
 
         # use the lemma of the words or not
-        self.use_lemma = self.component_config["use_lemma"]
+        self.use_lemma = self._config["use_lemma"]
 
     def _load_vocabulary_params(self) -> None:
-        self.OOV_token = self.component_config["OOV_token"]
+        self.OOV_token = self._config["OOV_token"]
```

Comment:
I think it would be cleaner if we pass "oov_tokens" and "oov_words" to the `init` and inside `init` use `_load_vocabulary_params` only if both are None (and otherwise set the oov words and tokens to the passed parameters). -- And if not, it would be cool to at least leave some notes somewhere that what happens here does not destroy the loaded oov words/tokens.

Reply:
done